File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/models/StarshipList.java

package com.dani_chuks.andeladeveloper.starwars.data.models;

import com.dani_chuks.andeladeveloper.starwars.data.models.entities.Starship;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class StarshipList {

    @SerializedName("count")
    @Expose
    private Integer count;
    @SerializedName("next")
    @Expose
    private String next;
    @SerializedName("previous")
    @Expose
    private Object previous;
    @SerializedName("results")
    @Expose
    private List<Starship> starship = null;

    public Integer getCount() {
        return count;
    }

    public void setCount(Integer count) {
        this.count = count;
    }

    public String getNext() {
        return next;
    }

    public void setNext(String next) {
        this.next = next;
    }

    public Object getPrevious() {
        return previous;
    }

    public void setPrevious(Object previous) {
        this.previous = previous;
    }

    public List<Starship> getStarship() {
        return starship;
    }

    public void setStarship(List<Starship> starship) {
        this.starship = starship;
    }

}

File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/models/StarshipList.kt
package com.dani_chuks.andeladeveloper.starwars.data.models

import com.dani_chuks.andeladeveloper.starwars.data.models.entities.Starship
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class StarshipList {

    @SerializedName("count")
    @Expose
    var count: Int? = null
    @SerializedName("next")
    @Expose
    var next: String? = null
    @SerializedName("previous")
    @Expose
    var previous: Any? = null
    @SerializedName("results")
    @Expose
    var starship: List<Starship>? = null

}
